### PR TITLE
fix836_getLoadedLibraries2 (second effort at PR).

### PR DESCRIPTION
**Pull request by clanmills**

### CODE REVIEW

File: src/futils.cpp [hunk 1]
```diff
@@ -502,4 +493,71 @@ namespace Exiv2 {
...
+            }
+        }
+        // free resources
+        if ( files    ) procstat_freefiles(procstat, files);
+        if ( procs    ) procstat_freeprocs(procstat, procs);
+        if ( procstat ) procstat_close    (procstat);
+
+    #elif defined(__unix__)
+        // read file /proc/self/maps which has a list of files in memory
+        std::ifstream maps("/proc/self/maps",std::ifstream::in);
```
Comment (piponazo) [inline src/futils.cpp]: You will need to include the header `<fstream>`, right now the linux jobs are failing with this error:

```
/home/travis/build/Exiv2/exiv2/src/futils.cpp: In function ‘std::vector<std::__cxx11::basic_string<char> > Exiv2::getLoadedLibraries()’:
/home/travis/build/Exiv2/exiv2/src/futils.cpp:551:27: error: variable ‘std::ifstream maps’ has initializer but incomplete type
         std::ifstream maps("/proc/self/maps",std::ifstream::in);
```

File: src/version.cpp [hunk 2]
```diff
@@ -412,7 +308,7 @@ void Exiv2::dumpLibraryInfo(std::ostream& os,const exv_grep_keys_t& keys)
     output(os,keys,"curl"          , use_curl);
     if ( libs.begin() != libs.end() ) {
         output(os,keys,"executable" ,*libs.begin());
-        for ( Exiv2::StringVector_i lib = libs.begin()+1 ; lib != libs.end() ; ++lib )
+        for ( std::vector<std::string>::iterator lib = libs.begin()+1 ; lib != libs.end() ; ++lib )
```
Comment (piponazo) [inline src/version.cpp]: Here we could use `auto` for not typing the long iterator type:

```cpp
for (const auto& lib = libs.begin()+1 ; lib != libs.end() ; ++lib )
```

Comment (clanmills) [reply src/version.cpp]: That doesn't compile on the Mac with Clang.

**Comment by piponazo** [comment]
Windows builds are failing with this error:

```
..\src\futils.cpp(519): error C3861: 'lengthof': identifier not found
..\src\futils.cpp(522): error C3861: 'lengthof': identifier not found
```

And the Fedora_MinGW one with this one:

```
src/futils.cpp:518:61: error: 'lengthof' was not declared in this scope
         if ( EnumProcessModules(GetCurrentProcess(),handles,lengthof(handles),&cbNeeded)) {
```

File: src/futils.cpp [hunk 3]
```diff
@@ -502,4 +494,71 @@ namespace Exiv2 {
...
+    {
+        std::vector<std::string>  libs ;
+        std::set<std::string>     paths;
+        std::string               path ;
+
+    #if defined(WIN32) || defined(__CYGWIN__) || defined(__MINGW__)
+        // enumerate loaded libraries and determine path to executable
+        HMODULE handles[200];
+        DWORD   cbNeeded;
+        if ( EnumProcessModules(GetCurrentProcess(),handles,lengthof(handles),&cbNeeded)) {
```
Comment (D4N) [inline src/futils.cpp]:
```suggestion
        if ( EnumProcessModules(GetCurrentProcess(),handles,EXV_COUNTOF(handles),&cbNeeded)) {
```

File: src/futils.cpp [hunk 4]
```diff
@@ -502,4 +494,71 @@ namespace Exiv2 {
...
+            char szFilename[_MAX_PATH];
+            for ( DWORD h = 0 ; h < cbNeeded/sizeof(handles[0]) ; h++ ) {
+                GetModuleFileNameA(handles[h],szFilename,lengthof(szFilename)) ;
```
Comment (D4N) [inline src/futils.cpp]:
```suggestion
                GetModuleFileNameA(handles[h],szFilename,EXV_COUNTOF(szFilename)) ;
```

**Review by D4N — COMMENTED** [review]
Replacing `lengthof` with `EXV_COUNTOF` should do the trick.

File: src/futils.cpp (hunk 4 above)
Comment (clanmills) [reply src/futils.cpp]: You're right Dan.  That's what I should have done.  I added macro lengthof() to src/futils.cpp

**Review by piponazo — APPROVED** [review]

**Pull request merged**